app/resources/v1/streets.js

```diff
@@ -35,168 +35,189 @@ exports.post = function (req, res) {
     street.creator_ip = requestIp(req)
   }
 
-  const handleCreateStreet = function (err, s) {
-    if (err) {
+  const makeNamespacedId = async function () {
+    try {
+      if (street.creator_id) {
+        const row = await User.findByIdAndUpdate(street.creator_id,
+          { $inc: { 'last_street_id': 1 } },
+          { new: true, upsert: true })
+        const namespacedId = (row) ? row.last_street_id : null
+        if (!namespacedId) {
+          throw new Error(ERRORS.INTERNAL_ERROR)
+        }
+        return namespacedId
+      } else {
+        const row = await Sequence.findByIdAndUpdate('streets',
+          { $inc: { 'seq': 1 } },
+          { new: true, upsert: true })
+        const namespacedId = (row) ? row.seq : null
+        if (!namespacedId) {
+          throw new Error(ERRORS.INTERNAL_ERROR)
+        }
+        return namespacedId
+      }
+    } catch (err) {
       logger.error(err)
-      res.status(500).send('Could not create street.')
-      return
+      throw new Error(ERRORS.INTERNAL_ERROR)
     }
+  } // END function - makeNamespacedId
+
+  const saveStreet = async function () {
+    try {
+      if (body && body.originalStreetId) {
+        const origStreet = await Street.findOne({ id: body.originalStreetId })
+        if (!origStreet || origStreet.status === 'DELETED') {
+          throw new Error(ERRORS.STREET_NOT_FOUND)
+        }
 
-    s.asJson(function (err, streetJson) {
+        street.original_street_id = origStreet
+        const namespacedId = await makeNamespacedId()
+        street.namespaced_id = namespacedId
+        return street.save()
+      } else {
+        const namespacedId = await makeNamespacedId()
+        street.namespaced_id = namespacedId
+        return street.save()
+      }
```

Comment:
Similarly to comments above, this can be refactored to have smaller try/catch block and less repeated lines of code.